engine/gamesys/src/gamesys/components/comp_sound.cpp

```diff
@@ -184,7 +187,7 @@ namespace dmGameSystem
             dmLogError("Error deleting sound: (%d)", r);
             return dmGameObject::UPDATE_RESULT_UNKNOWN_ERROR;
         }
-        else if (entry.m_Listener.m_Fragment != 0x0 && entry.m_LuaCallback)
+        else if (entry.m_Listener.m_Fragment != 0x0 && entry.m_ShouldDispatchEvents)
```

Comment:
Should the fragment be checked when starting the sound?

Reply:
Also, in your previous check, you had:
```
else if (entry.m_Listener.m_Fragment != 0x0 && (entry.m_LuaCallback || (entry.m_isMessage && entry.m_PlayId != dmSound::INVALID_PLAY_ID)))
```
where you checked if the m_LuaCallback was set, then it should dispatch?
I don't see that in the new code? Is this intentional?

Reply:
>where you checked if the m_LuaCallback was set, then it should dispatch? 

I explained it below

Reply:
>Should the fragment be checked when starting the sound?

this is something I didn't touch and I think it not something we need at all since we get `m_Sender` . Maybe it's some kind of protection from situations when message was sent and right after that go removed. not sure, since it old logic which isn't related to task. But if you think it's important to change, I can change it.